### PR TITLE
chore: Update installation requirements in CONTRIBUTING.md

### LINKED ISSUE #27846: Can't build v13.0.0+ from source without ```g++``` and ```make```
### Current behavior

I ran ```npm i -g yarn```
I ran ```git clone https://github.com/cypress-io/cypress```
I ran ```cd cypress```
I ran ```yarn```

This works for versions v12.17.4 and below, but fails for versions v13.0.0 and up

#### Without ```g++``` installed
```
make: g++: No such file or directory
```

#### Without ```make``` installed
```
Error: not found: make
```

### Cypress Version

develop - https://github.com/cypress-io/cypress/commit/2247ffd79b226a7fe00dd3252020becf2c22b110

### Node version

v18.18.0

### Package Manager

yarn

### Package Manager Version

1.22.19

### Operating system

Linux

### Operating System Version

- **Operating System:** Debian GNU/Linux 11 (bullseye)
- **Kernel:** Linux 5.15.124-20260-g158060cabb35
- **Architecture:** x86-64

### Non-documented fix

Installing ```g++``` and ```make``` packages seem to be required as of v13.0.0 in order to build Cypress from source - this was not the case with v12.17.4 and earlier versions

### Other

If the above packages are intended to be required now, the [CONTRIBUTING file](https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md) should be updated to state this

### PR DESCRIPTION
**Pull request by Crustum7**
- Closes #27846 

### Additional details

CMake is needed to set up the developer environment

### Steps to test

na

### How has the user experience changed?

na

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by MikeMcC399** [comment]
@Crustum7 

- Thanks for picking this up!
- I think you need to change the commit title in https://github.com/cypress-io/cypress/pull/27878/commits/73d40587d893d334545c2b0236a666f6e8c012eb which says "Solves issue-27846" and this is not a semantic commit message. I would suggest: `docs: update installation requirements in contributing.md` instead.

**Review by alexsch01 — CHANGES_REQUESTED** [review]
instead of ```on Unix```, it should be ```on Debian-based systems```

**Comment by alexsch01** [comment]
@Crustum7 installing just ```cmake``` didn't fix the build issue

**Comment by MikeMcC399** [comment]
@alexsch01 

> installing just `cmake` didn't fix the build issue

Is there something missing if you followed the prerequisites in https://github.com/nodejs/node-gyp#on-unix?

**Comment by Crustum7** [comment]
@alexsch01 well as @MikeMcC399 mentioned [here](https://github.com/cypress-io/cypress/issues/27846#issuecomment-1731126408), `make` and `build-essential` are indirectly required through node-gyp so they should not need to be included here as well. I can change to "on Debian-based systems" though if you want

**Comment by alexsch01** [comment]
I made a PR that supersedes this one: https://github.com/cypress-io/cypress/pull/27879
```g++``` and ```make``` were not required to build Cypress before v13.0.0

**Comment by MikeMcC399** [comment]
If `cmake` is not installed then `yarn install` fails at:

1. rebuild of `cpu-features@0.0.2`
2. ~~`electron-rebuild -o better-sqlite3`~~

so I support adding it explicitly to the list of requirements to build Cypress. My understanding of reading https://github.com/nodejs/node-gyp#on-unix is that it is not specifically mentioned there. If I have misinterpreted that, then I'd be grateful for an explanation.

- Edit: the `electron-rebuild` deprecations are a separate issue: https://github.com/cypress-io/cypress/issues/27891

---
https://github.com/nodejs/node-gyp#on-unix

### On Unix

   * [A supported version of Python](https://devguide.python.org/versions/)
   * `make`
   * A proper C/C++ compiler toolchain, like [GCC](https://gcc.gnu.org)

**Review by MikeMcC399 — APPROVED** [review]
LGTM

**Comment by MikeMcC399** [comment]
@alexsch01 

> I made a PR that supersedes this one: #27879 `g++` and `make` were not required to build Cypress before v13.0.0

I believe that is unnecessary, since  [CONTRIBUTING > Requirements](https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#requirements) refers to 
 > [`python`](https://www.python.org/downloads/) (since we use `node-gyp`. See their [repo](https://github.com/nodejs/node-gyp) for Python version requirements.) which refers to

- https://github.com/nodejs/node-gyp and this lists `make` and `gcc`

These generic instructions are admittedly difficult to follow. Perhaps there should be a cook-book with some explicit instructions for operating systems commonly used to build Cypress, like for instance Ubuntu 22.04 LTS?

I would in any case support merging this PR to simply add `cmake` to the list of requirements and then separately work on the issue about making the instructions easier to understand and use.

**Comment by alexsch01** [comment]
```cmake``` isn't required for the ```yarn``` command to successfuly install Cypress

```g++``` and ```make``` are required on Linux

---

> A proper C/C++ compiler toolchain, like [GCC](https://gcc.gnu.org)

**GCC** is not enough for Cypress to build successfully

---

```build-essential``` package on debian solved my building issues because it includes ```g++``` and ```make```

**Comment by MikeMcC399** [comment]
@alexsch01 

Thanks for your research and explanation. I will check again on a clean install of Ubuntu if `build-essential` is enough. I was getting errors without `cmake`, however I may have overlooked something. According to https://packages.ubuntu.com/jammy/build-essential it contains `g++`, `gcc` and `make` so perhaps this would be a good package to explicitly recommend for Debian-based systems?

**Comment by alexsch01** [comment]
> @alexsch01
> 
> Thanks for your research and explanation. I will check again on a clean install of Ubuntu if `build-essential` is enough. I was getting errors without `cmake`, however I may have overlooked something. According to https://packages.ubuntu.com/jammy/build-essential it contains `g++`, `gcc` and `make` so perhaps this would be a good package to explicitly recommend for Debian-based systems?

My [PR](https://github.com/cypress-io/cypress/pull/27879) has g++ (this includes gcc) and make required

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -207,6 +207,7 @@ You must have the following installed on your system to contribute locally:
 - [`Node.js`](https://nodejs.org/en/) (See the root [.node-version](.node-version) file for the required version. You can find a list of tools on [node-version-usage](https://github.com/shadowspawn/node-version-usage) to switch the version of [`Node.js`](https://nodejs.org/en/) based on [.node-version](.node-version).)
 - [`yarn`](https://yarnpkg.com/en/docs/install)
 - [`python`](https://www.python.org/downloads/) (since we use `node-gyp`. See their [repo](https://github.com/nodejs/node-gyp) for Python version requirements.)
+- [`cmake`](https://cmake.org/) (`sudo apt install cmake` on Debian-based systems)
```
Comment (MikeMcC399) [inline CONTRIBUTING.md]:
```suggestion
  - Note for Debian-based systems: `python` is pre-installed.<br>`sudo apt install g++ make cmake` meets the additional requirements to run `node-gyp` in the context of building Cypress from source.
```

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
The requirements for running node-gyp are supposed to be covered in the [requirements section](https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#requirements) referring to `python`. In practice the instructions are too vague and too open to interpretation, apart from the explicit mention of `make`, so I suggest to make the text look like this, with an note especially for Debian-based systems (including Ubuntu):

- [`python`](https://www.python.org/downloads/) (since we use `node-gyp`. See their [repo](https://github.com/nodejs/node-gyp) for Python version requirements.)
  - Note for Debian-based systems: `python` is pre-installed.<br>`sudo apt install g++ make cmake` meets the additional requirements to run `node-gyp` in the context of building Cypress from source.

**Review by MikeMcC399 — APPROVED** [review]
Thank you very much @Crustum7! 👍🏻 

I'm confident that this is correct advice now and should help anybody trying to set up a new Debian-based system to build Cypress from source.

It would certainly have helped me when I first did it last year and was struggling with trial-and-error methods!

**Comment by MikeMcC399** [comment]
@Crustum7 

Just for future reference if you create any new PR, there are some instructions in https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#pull-requests about naming the branch. It's not good practice to make edits in the default `develop` branch, even if you are working from a fork. Unfortunately you can't change the source branch name of an open PR, you can only change the target branch name, so don't worry about this for the moment. Just leave it as it is!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -207,6 +207,7 @@ You must have the following installed on your system to contribute locally:
 - [`Node.js`](https://nodejs.org/en/) (See the root [.node-version](.node-version) file for the required version. You can find a list of tools on [node-version-usage](https://github.com/shadowspawn/node-version-usage) to switch the version of [`Node.js`](https://nodejs.org/en/) based on [.node-version](.node-version).)
 - [`yarn`](https://yarnpkg.com/en/docs/install)
 - [`python`](https://www.python.org/downloads/) (since we use `node-gyp`. See their [repo](https://github.com/nodejs/node-gyp) for Python version requirements.)
+  - Note for Debian-based systems: `python` is pre-installed.<br>`sudo apt install g++ make cmake` meets the additional requirements to run `node-gyp` in the context of building Cypress from source.
```
Comment (MikeMcC399) [inline CONTRIBUTING.md:210]: ~~Sorry for this. One more correction, pointed out by @alexsch01: If `cmake` is installed, it also installs `make`.~~

Correction to correction:

`cmake` only
`Recommends: gcc, make`
so `cmake`  should stay in the install command.

Comment (alexsch01) [reply CONTRIBUTING.md:210]: Actually nope, make should still be in the install command

Sorry for the confusion

**Comment by alexsch01** [comment]
The current commit looks good to me

File: CONTRIBUTING.md (hunk 2 above)
Comment (MikeMcC399) [reply CONTRIBUTING.md:210]: I've hidden the latest suggestion. I'm not an `apt` expert. I have just been checking that the instructions work on Ubuntu.

**Review by alexsch01 — APPROVED** [review]

**Review by nagash77 — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

**Comment by MikeMcC399** [comment]
Just as a sanity cross-check, the following instructions will build Cypress on a freshly-installed Ubuntu `22.04`  system with no other prerequisites installed:

```bash
sudo apt install git g++ cmake make curl
curl -L https://bit.ly/n-install | bash
 . ~/.bashrc
cd ~
git clone https://github.com/cypress-io/cypress
cd cypress
n auto
npm install yarn -g
yarn
```

**Pull request merged**